mypy/fastparse.py

```diff
@@ -762,6 +765,39 @@ def visit_ExtSlice(self, n: ast35.ExtSlice) -> Node:
     def visit_Index(self, n: ast35.Index) -> Node:
         return self.visit(n.value)
 
+    # PEP 492 nodes: 'async def', 'await', 'async for', 'async with'.
```

Comment:
That makes sense -- moved them back (except moved AsyncFunctionDef up so
it's better paired with FunctionDef).

Also did a bunch of work on checker.py to remove the remaining XXX comments
(originating in this PR).